doc/references/taxonomy-parser.md

```diff
@@ -0,0 +1,161 @@
+   line = re.sub(r"\\,", "\\‚", line)
+   ```
+1. removes parenthesis for roman numerals
+   ```python
+   line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+   ```
+
+## Normalized version of a string (of a tag)
+
+The *normalized version* of a string is the operation
+we will apply that to tag to obtain the tagid.
+
+It depends on the language.
+
+Obtain NFC form of the string. In python:
+```python
+string = unicodedata.normalize("NFC", my_input)
+```
+
+Then look at `string_normalization_for_lang` in `Config_off.pm` [^string_config]
+to know if following operations applies, based on language code:
+* lowercase
+* removing accents
+
+If lowercase applies and string is not a UUID, that is it does not match regexp `r"^[a-z\-]+\.[a-zA-Z0-9-_]{8}[a-zA-Z0-9-_]+$"`,
+then lowercase it. And also transform "." to "-".
+
+Replace every special character (ascii 0 to 27) and zero width space with "-":
+```python
+string = re.sub(r"[\u0000-\u0027\u200b]", "-", string)
+```
+
+Replace html escaped chars by "-":
+```python
+string = re.sub(r"&\w+;", "-", string)
+```
+
+Replace some weirds chars by "-" [^strech_weird_chars]:
+```
+string = re.sub(r"[\s!"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", string)
+```
+
+If removing accents applies, use `unidecode.unidecode` to remove them.
+
+Finally, replace consecutive "-" by only one minus, and strip "-":
+```python
+string = re.sub(r"-+", "-", string)
+string = string.strip("-")
+```
+
+[^string_config]: of course you would have copied it to a local config file, or a dict in your code.
+
+[^strech_weird_chars]: (stretch goal: use [unicode category](https://www.compart.com/en/unicode/category) (through unicodedata module) to remove all unwanted category)
+
+# Processing
+
+## Harvest data
+
+1. store the header in a node with id `__header__`
+2. store each block of *stopwords line* in a node with id `stopwords:IDX` where `IDX` is the number of preceding *stopwords lines* in the file, and store inside:
+   * every tag and tagid with language code included
+3. store each block of *global synonyms line* in a node with id `synonyms:IDX` where `IDX` is the number of preceding *global synonyms lines* in the file, , and store inside:
+   * every tag and tagid with language code included
+2. For each entry block, create a node with id `LC:canonical_tagid`
+   store inside:
+   * every *property value* (a simple json)
+   * every *parent tag* with their *tagid*, language code included (this is temporary)
+   * every *tags*, with their *tagid*, language code included
+
+For every node, also store:
+* a specific `block_index` property which stores the number of blocks before this one in the file
+* a specific `block_subindex` property which is always 0
+  (such that if we want to insert a new block beneath this block,
+  later on,
+  we could insert it with same block_index but block_subindex of 1)
+* preceding *comments lines* in a `preceeding_lines` property
```

Comment:
As far as I can understand, the *preceeding_lines* id stores the number of comment lines present before the current block. But, a way for storing comments isn't mentioned in the document. 

Would it be optimal to add a tag called *comment*, with its value as the comment line?